bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttBrokerConnectionTests.java

```diff
@@ -0,0 +1,185 @@
+            MqttBrokerConnection ignored = new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
+            fail("Configuration exception expected: No valid url protocol");
+        } catch (ConfigurationException e) {
+        }
+        // Test an invalid name
+        try {
+            @SuppressWarnings("unused")
+            MqttBrokerConnection ignored = new MqttBrokerConnection(" ", "tcp://123.123.123.123", false);
+            fail("Configuration exception expected: No valid name");
+        } catch (ConfigurationException e) {
+        }
+    }
+
+    @Test
+    public void messageConsumerTests() {
+        final String url = "tcp://123.123.123.123";
+        final String name = "TestName12@!";
+        MqttBrokerConnection a;
+        try {
+            a = new MqttBrokerConnection(name, url, false);
+        } catch (ConfigurationException e) {
+            fail();
+            return;
+        }
+        assertFalse(a.isConsumers());
+        MqttMessageSubscriber subscriber = new MqttMessageSubscriber() {
+            @Override
+            public void processMessage(String topic, byte[] payload) {
+
+            }
+
+            @Override
+            public String getTopic() {
+                return "topic";
+
+            }
+        };
+        try {
+            a.addConsumer(subscriber);
+        } catch (MqttException e) {
+            fail("addConsumer failed");
+        }
+        assertTrue(a.isConsumers());
+        a.removeConsumer(subscriber);
+        assertFalse(a.isConsumers());
+    }
+
+    @Test
+    public void connectionObserverTests() {
+        final String url = "tcp://123.123.123.123";
+        final String name = "TestName12@!";
+        MqttBrokerConnection a;
+        try {
+            a = new MqttBrokerConnection(name, url, false);
+        } catch (ConfigurationException e) {
+            fail();
+            return;
+        }
+        assertFalse(a.isConnectionObservers());
+        isConnected = true;
+        MqttConnectionObserver connectionObserver = new MqttConnectionObserver() {
+            @Override
+            public void setConnected(boolean connected) {
+                isConnected = connected;
+            }
+        };
+        a.addConnectionObserver(connectionObserver);
+        // Adding a connection observer should immediately call its setConnected() method.
+        assertFalse(isConnected);
+
+        assertTrue(a.isConnectionObservers());
+        a.removeConnectionObserver(connectionObserver);
+        assertFalse(a.isConnectionObservers());
+    }
+
+    @Test
+    public void setterGetterTests() {
+        final String url = "tcp://123.123.123.123";
+        final String name = "TestName12@!";
+        MqttBrokerConnection a;
+        try {
+            a = new MqttBrokerConnection(name, url, false);
+        } catch (ConfigurationException e) {
+            fail();
+            return;
+        }
+        assertTrue("URL getter", a.getUrl().equals(url));
```

Comment:
Please use `assertEquals(url, a.getUrl());` or even better `assertThat(a.getUrl(), is(url));` here. Please have a look at the other assertions where semantically more appropriate methods are available.